vendor/engines/split_accounts/app/services/split_accounts/account_builder_extension.rb

```diff
@@ -0,0 +1,22 @@
+module SplitAccounts
+
+  # Includes AccountBuilder extensions for the SplitAccounts::SplitAccount
+  # account type.
+  module AccountBuilderExtension
+    extend ActiveSupport::Concern
+
+    # Dynamically called by `build_subclass` if the account_type is NufsAccount.
+    def build_split_accounts_split_account(account, params)
+      account = build_split_account_expires_at(account, params)
+      account
+    end
+
+
+    # Sets `expires_at` for SplitAccounts::SplitAccount only.
+    def build_split_account_expires_at(account, params)
+      account.expires_at = Time.zone.now + 50.years
+      account
+    end
```

Comment:
The expiration date of a split account should be the earliest of all the component accounts' expirations.